### ISSUE TITLE
[BUG] Chromium crash with '!observations_.IsObservingAnySource(). Some render process hosts were not destroyed early enough'

### ISSUE THREAD

**Issue by gadezist** [microsoft/playwright#16748]
**Context:**
- Playwright Version: 1.25.0
- Operating System:  Ubuntu 22.04.1 LTS
- Node.js version: 18.7.0
- Browser: Chromium

**Describe the bug**

This is a continuation of issue #16677.
I created a repo with the code that reproduce the bug. Here is the link - https://github.com/gadezist/playwright_bug/tree/master

If these tests are run in parallel, everything works fine, but if not, the test crashes with the following error
![image](https://user-images.githubusercontent.com/39760138/186140566-f1d0a84d-d585-4eab-88fd-2462c1436fa3.png)

But until version 1.25.0 everything worked fine

**Comment by yury-s** [comment]
It fails because of a browser crash, running with `DEBUG=pw:browser` shows the following trace:

```
  pw:browser [pid=74518][err] [74518:259:0823/131011.561696:FATAL:profile_destroyer.cc(266)] Check failed: !observations_.IsObservingAnySource(). Some render process hosts were not destroyed early enough! +4s
  pw:browser [pid=74518][err] 0   Chromium Framework                  0x00000001172b36d0 base::debug::CollectStackTrace(void**, unsigned long) + 28 +0ms
  pw:browser [pid=74518][err] 1   Chromium Framework                  0x000000011721c96c base::debug::StackTrace::StackTrace() + 24 +0ms
  pw:browser [pid=74518][err] 2   Chromium Framework                  0x000000011722e390 logging::LogMessage::~LogMessage() + 156 +0ms
  pw:browser [pid=74518][err] 3   Chromium Framework                  0x000000011722ef5c logging::LogMessage::~LogMessage() + 12 +1ms
  pw:browser [pid=74518][err] 4   Chromium Framework                  0x0000000117217d9c logging::CheckError::~CheckError() + 36 +0ms
  pw:browser [pid=74518][err] 5   Chromium Framework                  0x0000000116f87960 ProfileDestroyer::~ProfileDestroyer() + 228 +0ms
  pw:browser [pid=74518][err] 6   Chromium Framework                  0x0000000116f87db8 ProfileDestroyer::~ProfileDestroyer() + 12 +0ms
  pw:browser [pid=74518][err] 7   Chromium Framework                  0x0000000113e0c824 base::internal::Invoker<base::internal::BindState<void (net::(anonymous namespace)::DnsHTTPAttempt::*)(), base::WeakPtr<net::(anonymous namespace)::DnsHTTPAttempt>>, void ()>::RunOnce(base::internal::BindStateBase*) + 132 +0ms
  pw:browser [pid=74518][err] 8   Chromium Framework                  0x00000001172a59f4 base::OneShotTimer::FireNow() + 52 +0ms
  pw:browser [pid=74518][err] 9   Chromium Framework                  0x0000000117289490 base::DefaultDelayedTaskHandleDelegate::RunTask(base::OnceCallback<void ()>) + 48 +0ms
  pw:browser [pid=74518][err] 10  Chromium Framework                  0x0000000114425e00 void base::internal::FunctorTraits<void (media::PipelineImpl::RendererWrapper::*)(base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>), void>::Invoke<void (media::PipelineImpl::RendererWrapper::*)(base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>), base::WeakPtr<media::PipelineImpl::RendererWrapper>, base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>>(void (media::PipelineImpl::RendererWrapper::*)(base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>), base::WeakPtr<media::PipelineImpl::RendererWrapper>&&, base::OnceCallback<void (media::TypedStatus<media::PipelineStatusTraits>)>&&) + 132 +0ms
  pw:browser [pid=74518][err] 11  Chromium Framework                  0x000000011727108c base::TaskAnnotator::RunTaskImpl(base::PendingTask&) + 272 +0ms
  pw:browser [pid=74518][err] 12  Chromium Framework                  0x0000000117285524 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWorkImpl(base::sequence_manager::LazyNow*) + 644 +0ms
  pw:browser [pid=74518][err] 13  Chromium Framework                  0x0000000117285164 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::DoWork() + 100 +0ms
  pw:browser [pid=74518][err] 14  Chromium Framework                  0x00000001172c3ce4 base::MessagePumpCFRunLoopBase::RunWork() + 104 +0ms
  pw:browser [pid=74518][err] 15  Chromium Framework                  0x00000001172be9a8 base::mac::CallWithEHFrame(void () block_pointer) + 16 +0ms
  pw:browser [pid=74518][err] 16  Chromium Framework                  0x00000001172c3474 base::MessagePumpCFRunLoopBase::RunDelayedWorkTimer(__CFRunLoopTimer*, void*) + 68 +0ms
  pw:browser [pid=74518][err] 17  CoreFoundation                      0x00000001af09f964 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32 +0ms
  pw:browser [pid=74518][err] 18  CoreFoundation                      0x00000001af09f54c __CFRunLoopDoTimer + 1064 +0ms
  pw:browser [pid=74518][err] 19  CoreFoundation                      0x00000001af09efd4 __CFRunLoopDoTimers + 392 +0ms
  pw:browser [pid=74518][err] 20  CoreFoundation                      0x00000001af083a64 __CFRunLoopRun + 1936 +0ms
  pw:browser [pid=74518][err] 21  CoreFoundation                      0x00000001af082b34 CFRunLoopRunSpecific + 600 +0ms
  pw:browser [pid=74518][err] 22  HIToolbox                           0x00000001b7cc2338 RunCurrentEventLoopInMode + 292 +0ms
  pw:browser [pid=74518][err] 23  HIToolbox                           0x00000001b7cc20b4 ReceiveNextEventCommon + 564 +0ms
  pw:browser [pid=74518][err] 24  HIToolbox                           0x00000001b7cc1e68 _BlockUntilNextEventMatchingListInModeWithFilter + 72 +0ms
  pw:browser [pid=74518][err] 25  AppKit                              0x00000001b1bea51c _DPSNextEvent + 860 +0ms
  pw:browser [pid=74518][err] 26  AppKit                              0x00000001b1be8e14 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1328 +0ms
  pw:browser [pid=74518][err] 27  Chromium Framework                  0x0000000116e2de7c __71-[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:]_block_invoke + 64 +0ms
  pw:browser [pid=74518][err] 28  Chromium Framework                  0x00000001172be9a8 base::mac::CallWithEHFrame(void () block_pointer) + 16 +0ms
  pw:browser [pid=74518][err] 29  Chromium Framework                  0x0000000116e2dde0 -[BrowserCrApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 128 +0ms
  pw:browser [pid=74518][err] 30  AppKit                              0x00000001b1bdafe0 -[NSApplication run] + 596 +0ms
  pw:browser [pid=74518][err] 31  Chromium Framework                  0x00000001172c4494 base::MessagePumpNSApplication::DoRun(base::MessagePump::Delegate*) + 312 +0ms
  pw:browser [pid=74518][err] 32  Chromium Framework                  0x00000001172c2fd4 base::MessagePumpCFRunLoopBase::Run(base::MessagePump::Delegate*) + 216 +0ms
  pw:browser [pid=74518][err] 33  Chromium Framework                  0x0000000117285cf4 base::sequence_manager::internal::ThreadControllerWithMessagePumpImpl::Run(bool, base::TimeDelta) + 320 +0ms
  pw:browser [pid=74518][err] 34  Chromium Framework                  0x000000011725682c base::RunLoop::Run(base::Location const&) + 436 +0ms
  pw:browser [pid=74518][err] 35  Chromium Framework                  0x0000000114e111f0 content::BrowserMainLoop::RunMainMessageLoop() + 180 +0ms
  pw:browser [pid=74518][err] 36  Chromium Framework                  0x0000000114e128f0 content::BrowserMainRunnerImpl::Run() + 24 +0ms
  pw:browser [pid=74518][err] 37  Chromium Framework                  0x0000000114e0efc0 content::BrowserMain(content::MainFunctionParams) + 200 +0ms
  pw:browser [pid=74518][err] 38  Chromium Framework                  0x0000000116d9eda0 content::RunBrowserProcessMain(content::MainFunctionParams, content::ContentMainDelegate*) + 156 +0ms
  pw:browser [pid=74518][err] 39  Chromium Framework                  0x0000000116d9ffb8 content::ContentMainRunnerImpl::RunBrowser(content::MainFunctionParams, bool) + 456 +0ms
  pw:browser [pid=74518][err] 40  Chromium Framework                  0x0000000116d9fd7c content::ContentMainRunnerImpl::Run() + 820 +0ms
  pw:browser [pid=74518][err] 41  Chromium Framework                  0x0000000116d9e840 content::RunContentProcess(content::ContentMainParams, content::ContentMainRunner*) + 2448 +0ms
  pw:browser [pid=74518][err] 42  Chromium Framework                  0x0000000116d9e978 content::ContentMain(content::ContentMainParams) + 84 +0ms
  pw:browser [pid=74518][err] 43  Chromium Framework                  0x0000000113ceccdc ChromeMain + 256 +0ms
  pw:browser [pid=74518][err] 44  Chromium                            0x0000000104018938 main + 288 +0ms
  pw:browser [pid=74518][err] 45  dyld                                0x000000010414108c start + 520 +0ms
  pw:browser [pid=74518][err] Task trace: +0ms
  pw:browser [pid=74518][err] 0   Chromium Framework                  0x0000000116f874cc ProfileDestroyer::ProfileDestroyer(Profile*, std::Cr::set<content::RenderProcessHost*, std::Cr::less<content::RenderProcessHost*>, std::Cr::allocator<content::RenderProcessHost*>> const&) + 380 +0ms
  pw:browser [pid=74518][err] 1   Chromium Framework                  0x00000001197eb2f0 DevToolsBrowserContextManager::OnBrowserRemoved(Browser*) + 296 +0ms
  pw:browser [pid=74518][err] 2   Chromium Framework                  0x0000000118aed53c remote_cocoa::NativeWidgetNSWindowBridge::CloseWindow() + 472 +0ms
  pw:browser [pid=74518][err] 3   Chromium Framework                  0x0000000117c1a364 IPC::(anonymous namespace)::ChannelAssociatedGroupController::Accept(mojo::Message*) + 1092 +0ms
  pw:browser [pid=74518][err] 4   Chromium Framework                  0x0000000117426904 mojo::SimpleWatcher::Context::Notify(unsigned int, MojoHandleSignalsState, unsigned int) + 288 +0ms
  pw:browser [pid=74518][err] Crash keys: +0ms
  pw:browser [pid=74518][err]   "total-discardable-memory-allocated" = "8388608" +0ms
  pw:browser [pid=74518][err]   "gpu-glver" = "OpenGL ES 2.0.0 (ANGLE 2.1.19212 git hash: 57ccd974bad6)" +0ms
  pw:browser [pid=74518][err]   "gpu-generation-intel" = "0" +0ms
  pw:browser [pid=74518][err]   "gpu-vsver" = "1.00" +0ms
  pw:browser [pid=74518][err]   "gpu-psver" = "1.00" +0ms
  pw:browser [pid=74518][err]   "gpu-driver" = "76.3" +0ms
  pw:browser [pid=74518][err]   "gpu-devid" = "0x0000" +0ms

```

**Comment by yury-s** [comment]
The problem was  fixed upstream as part of https://bugs.chromium.org/p/chromium/issues/detail?id=1349150 (restricted access at the moment) and is already available in Chrome Canary, you can give it a try by setting `channel: 'chrome-canary'` in the playwright config. The fix should be rolled to playwright before 1.26 release.

**Comment by dgozman** [comment]
Fixed by the recent Chromium roll.
